packages/bot/src/components/flyout.jsx

```diff
@@ -0,0 +1,33 @@
+import React            from 'react';
+import PropTypes        from 'prop-types';
+import { connect }      from '../stores/connect';
+import                       '../assets/sass/scratch/_flyout.scss';
+
+class Flyout extends React.PureComponent {
+    render() {
+        const { is_visible, flyout_width, flyout_content } = this.props;
+
+        return (
+            <div
+                ref={node => this.node = node}
+                className={`flyout${!is_visible ? ' flyout--hidden' : ''}`}
+                style={{ width: `${flyout_width}px` }}
```

Comment:
@aaron-binary why we need inline styles? if we have to we can use css-variables.

Reply:
We don't know the width until we get the blocks that are added to the flyout. These blocks can be user generated ones in the case of custom procedures or variables.